helm/prometheus-meta-operator/templates/recording-rules/service-levels.rules.yml

```diff
@@ -50,6 +50,62 @@ spec:
         service: kubelet
       record: slo_target
 
+    # -- managed-apps
+    # record when pods of a daemonset with label "label_giantswarm_io_monitoring_basic_sli" are down
+    - expr: |
+        label_replace(kube_daemonset_status_number_unavailable
+        and on(daemonset,cluster_id,cluster_type,namespace, workload_name)
+        kube_daemonset_labels{label_giantswarm_io_monitoring_basic_sli='true'}, "service", "$1", "workload_name", "(.*)" )
+      labels:
+        class: MEDIUM
+        area: managed-apps
+      record: raw_slo_errors
+    # record when pods of a deployment with label "label_giantswarm_io_monitoring_basic_sli" are down
+    - expr: |
+        label_replace(kube_deployment_status_replicas_unavailable
+        and on(deployment,cluster_id,cluster_type,namespace, workload_name) 
+        kube_deployment_labels{label_giantswarm_io_monitoring_basic_sli='true'}, "service", "$1", "workload_name", "(.*)" )
+      labels:
+        class: MEDIUM
+        area: managed-apps
+      record: raw_slo_errors
+    # record when pods of a statefulset with label "label_giantswarm_io_monitoring_basic_sli" are down
+    - expr: |
+        label_replace(kube_statefulset_status_replicas - kube_statefulset_status_replicas_current
+        and on(statefulset,cluster_id,cluster_type,namespace, workload_name) 
+        kube_statefulset_labels{label_giantswarm_io_monitoring_basic_sli='true'}, "service", "$1", "workload_name", "(.*)" )
+      labels:
+        class: MEDIUM
+        area: managed-apps
+      record: raw_slo_errors
+    - expr: |
+        label_replace(kube_daemonset_status_desired_number_scheduled
+        and on(daemonset,cluster_id,cluster_type,namespace, workload_name)
+        kube_daemonset_labels{label_giantswarm_io_monitoring_basic_sli='true'}, "service", "$1", "workload_name", "(.*)" )
+      labels:
+        class: MEDIUM
+        area: managed-apps
+      record: raw_slo_requests
+    - expr: |
+        label_replace(kube_deployment_status_replicas
+        and on(deployment,cluster_id,cluster_type,namespace, workload_name) 
+        kube_deployment_labels{label_giantswarm_io_monitoring_basic_sli='true'}, "service", "$1", "workload_name", "(.*)" )
+      labels:
+        class: MEDIUM
+        area: managed-apps
+      record: raw_slo_requests
+    - expr: |
+        label_replace(kube_statefulset_status_replicas
+        and on(statefulset,cluster_id,cluster_type,namespace, workload_name) 
+        kube_statefulset_labels{label_giantswarm_io_monitoring_basic_sli='true'}, "service", "$1", "workload_name", "(.*)" )
+      labels:
+        class: MEDIUM
+        area: managed-apps
+      record: raw_slo_requests
+      # -- 99.5% availability
+    - expr: sum by (service) (raw_slo_errors{area="managed-apps"} - raw_slo_errors{area="managed-apps"}) + 1-0.995
+      record: slo_target
```

Comment:
That's sneaky :D

Reply:
:D It was the only thing I could come up with that would add all the services with the right `slo_target` without explicitly listing them.

Reply:
I actually don't get these 2 lines... what do you want to do here? Insert target for all existing services?

Reply:
It basically emits `slo_target` of 0.005 for all services from managed-apps area.

Reply:
Exactly. I just do this stupid math with substracting from itself to make sure it is 0 before adding the value I want.

Reply:
Can you document it in the rule?